vignettes/epipredict.Rmd

```diff
@@ -1,470 +1,565 @@
+      or [be defined in `{recipes}`](https://recipes.tidymodels.org/reference/index.html).
+      `four_week_ahead` has 5 steps; you can inspect them more closely by
+      running `hardhat::extract_recipe(four_week_ahead$epi_workflow)`.[^6]
+- `Model`: the actual model that does the fitting, given by a
+  `parsnip::model_spec`. `four_week_ahead` uses the default of
+  `parsnip::linear_reg()`, which is a `{parsnip}` wrapper for
+  `stats::lm()`. You can inspect the model more closely by running
+  `hardhat::extract_fit_recipe(four_week_ahead$epi_workflow)`.
+- `Postprocessor`: a collection of layers to be applied to the resulting
+  forecast. Layers are internal to this package. `four_week_ahead` just so happens to have
+  5 of as these well. You can inspect the layers more closely by running
+  `epipredict::extract_layers(four_week_ahead$epi_workflow)`.
+
+See the [Custom Epiworkflows vignette](custom_epiworkflows) for recreating and then
+extending `four_week_ahead` using the custom forecaster framework.
+
+## Mathematical description
+
+Let's look at the mathematical details of the model in more detail, using a minimal version of
+`four_week_ahead`:
+
+```{r, four_week_again}
+four_week_small <- arx_forecaster(
+  covid_case_death_rates |> filter(time_value <= forecast_date),
+  outcome = "death_rate",
+  predictors = c("case_rate", "death_rate"),
+  args_list = arx_args_list(
+    lags = list(c(0, 7, 14), c(0, 7, 14)),
+    ahead = 4 * 7,
+    quantile_levels = c(0.1, 0.25, 0.5, 0.75, 0.9)
+  )
+)
+hardhat::extract_fit_engine(four_week_small$epi_workflow)
+```
 
-Internally, we provide some simple functions to create reasonable forecasts.
-But ideally, a user could create their own forecasters by building up the
-components we provide. In other vignettes, we try to walk through some of these
-customizations.
+If $d_t$ is the death rate on day $t$ and $c_t$ is the case rate, then the model
+we're fitting is:
 
-To illustrate everything above, here is (roughly) the code for the
-`flatline_forecaster()` applied to the `case_rate`.
+$$
+d_{t+28} = a_0 + a_1 d_t + a_2 d_{t-7} + a_3 d_{t-14} + a_4 c_t + a_5 c_{t-7} + a_6 c_{t-14}.
+$$
 
-```{r}
-r <- epi_recipe(jhu) %>%
-  step_epi_ahead(case_rate, ahead = 7, skip = TRUE) %>%
-  update_role(case_rate, new_role = "predictor") %>%
-  add_role(all_of(key_colnames(jhu)), new_role = "predictor")
+For example, $a_1$ is `lag_0_death_rate` above, with a value of `r hardhat::extract_fit_engine(four_week_small$epi_workflow)$coefficients["lag_0_death_rate"] `,
+while $a_5$ is `r hardhat::extract_fit_engine(four_week_small$epi_workflow)$coefficients["lag_7_case_rate"] `.
 
-f <- frosting() %>%
-  layer_predict() %>%
-  layer_residual_quantiles() %>%
-  layer_add_forecast_date() %>%
-  layer_add_target_date() %>%
-  layer_threshold(starts_with(".pred"))
+The training data for fitting this linear model is constructed within the `arx_forecaster()` function by shifting a series
```

Comment:
rather than "fitting", you should say "estimating the parameters of".

Reading it here made me search out other instances above. Possibly worth a
discussion elsewhere.

Reply:
I'm trying to have the language mirror the actual function used